source/commands/user.cpp

```diff
@@ -1,13 +1,51 @@
 #include "commands.hpp"
 
+#define SPACE ' '
+#define NUL '\0'
+#define CR '\r'
+#define LF '\n'
+
+bool nonwhite(char c)
+{
+	if (c == SPACE)
+		return (false);
+	if (c == NUL)
+		return (false);
+	if (c == CR)
+		return (false);
+	if (c == LF)
+		return (false);
+	return (true);
+}
+
+bool checkUser(std::string usr)
+{
+	for (size_t x = 0; x < usr.length(); x ++)
+	{
+		if (nonwhite(usr[x]))
+			return (false);
+	}
+	return (true);
+}
+
 void cmd::user(const ft::Message& msg, ft::Client& client, ft::IRC& irc)
 {
 	(void)irc;
 	if (msg.parameters.size() != 4)
-		return;
+	{
+		client.sendmsg(ft::Message(":127.0.0.1 461 :Not enough parameters")); //ERR_NEEDMOREPARAMS
+		return ;
+	}
+	if (client.getUser().empty() == false || !client._raspberry)
```

Comment:
I don't think the `!client._raspberry` check is good here, since the check returns `ERR_ALREADYREGISTERED`, which could be misleading to some users.
```suggestion
	if (client.getUser().empty() == false)
```